src/main/java/org/jmxtrans/agent/ResultNameStrategyImpl.java

```diff
@@ -105,10 +105,12 @@
     public String getResultName(@Nonnull Query query, @Nonnull ObjectName objectName, @Nullable String key, @Nonnull String attribute) {
```

Comment:
Why did you change the existing `ResultNameStrategyImpl`? Aren't you adding another implementation with `JConsoleNameStrategyImpl`?

Don't we have a backward compatibility risk if we modify `ResultNameStrategyImpl`?